src/AutoRest.CSharp/Common/Utilities/StringExtensions.cs

```diff
@@ -45,19 +47,33 @@ public static string ToCleanName(this string name, bool camelCase = true)
             }
 
             bool upperCase = false;
+            int firstWordLength = 1;
```

Comment:
Why not set this correctly right here instead of constantly incrementing this?  We should be able to do a linear scan and store the first word length in one go.

Reply:
That could work, but initially I was finding a patch to this function instead of rewriting it. And we only need this when camelCase is false

Reply:
Can we add a follow up item to address this?